File: common/scanner/src/main/java/ru/tensor/sbis/scanner/view/decoration/GridSpacingItemDecoration.java
package ru.tensor.sbis.scanner.view.decoration;

import android.graphics.Rect;
import androidx.recyclerview.widget.RecyclerView;
import android.view.View;

/**
 * @author am.boldinov
 */
public class GridSpacingItemDecoration extends RecyclerView.ItemDecoration {

    private final int mSpanCount;
    private final int mSpacingInPx;

    public GridSpacingItemDecoration(int spanCount, int spacingInPx) {
        mSpanCount = spanCount;
        mSpacingInPx = spacingInPx;
    }

    @Override
    public void getItemOffsets(Rect outRect, View view, RecyclerView parent, RecyclerView.State state) {
        int position = parent.getChildAdapterPosition(view);
        int column = position % mSpanCount;

        outRect.left = mSpacingInPx - column * mSpacingInPx / mSpanCount;
        outRect.right = (column + 1) * mSpacingInPx / mSpanCount;

        if (position < mSpanCount) {
            outRect.top = mSpacingInPx;
        }
        outRect.bottom = mSpacingInPx;
    }
}

File: design/design_menu/src/main/java/ru/tensor/sbis/design/design_menu/quick_action_menu/GridSpacingItemDecoration.kt
package ru.tensor.sbis.design.design_menu.quick_action_menu

import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.RecyclerView

/**
 * Декоратор для установки вертикальных и горизонтальных отступов между элементами списка.
 *  @param spanCount количество столбцов в списке
 *  @param spacing отступ между элементами списка
 *
 * @author ra.geraskin
 */

class GridSpacingItemDecoration(private val spanCount: Int, private val spacing: Int) : RecyclerView.ItemDecoration() {

    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
        val position = parent.getChildAdapterPosition(view)
        val column = position % spanCount
        outRect.right = column * spacing / spanCount
        outRect.left = spacing - (column + 1) * spacing / spanCount
        if (position >= spanCount) {
            outRect.top = spacing
        }
    }

}
